src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1777,17 +1777,11 @@ private IEnumerable<TypeInferenceInfo> InferTypeInYieldStatement(YieldStatementS
                 var memberSymbol = GetDeclaredMemberSymbolFromOriginalSemanticModel(declaration);
 
                 var memberType = GetMemberType(memberSymbol, out _);
-
-                if (memberType is INamedTypeSymbol namedType)
-                {
-                    if (memberType.OriginalDefinition.SpecialType == SpecialType.System_Collections_Generic_IEnumerable_T ||
-                        memberType.OriginalDefinition.SpecialType == SpecialType.System_Collections_Generic_IEnumerator_T)
-                    {
-                        return SpecializedCollections.SingletonEnumerable(new TypeInferenceInfo(namedType.TypeArguments[0]));
-                    }
-                }
-
-                return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
+                // We don't care what the type is, as long as it has 1 type argument. This will work for IEnumerable, IEnumerator,
```

Comment:
```suggestion

                // We don't care what the type is, as long as it has 1 type argument. This will work for IEnumerable, IEnumerator,
```